src/riak_ql_ddl.erl

```diff
@@ -204,11 +204,11 @@ lk_to_pk(LKVals, Mod, ?DDL{local_key = #key_v1{ast = LKAst},
                            partition_key = PK}) ->
     KeyFields = [F || ?SQL_PARAM{name = [F]} <- LKAst],
     case {length(LKVals), length(KeyFields)} of
-        {_N, _N} ->
+        {_, _} ->
             LKPairs = lists:zip(KeyFields, LKVals),
             PKVals = [V || {_Type, V} <- make_key(Mod, PK, LKPairs)],
             {ok, PKVals};
-        {Got, Need} ->
+        {Got, Need} when Got =< 0, Need =< 0 ->
```

Comment:
Why those guards? The result of `length` is never less than 0, so the only case to get past the guard in your version is to have `Got` and `Need` == 0, whereas we want to check that `Got /= Need`.

There is no need for any additional guards in this clause.